controllers/gslb_controller_reconciliation.go

```diff
@@ -103,42 +104,6 @@ func (r *GslbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		Str("namespace", gslb.Namespace).
 		Interface("strategy", gslb.Spec.Strategy).
 		Msg("Resolved strategy")
-	// == Finalizer business ==
```

Comment:
Why did you move the "Finalizer business" down?

Afaiu we try to keep it as up as possible to terminate the reconciliation immediately if there GSLB is going to be deleted

Reply:
Agreed, I didn't want to, but the finalizer needs to [pass](https://github.com/k8gb-io/k8gb/pull/1774/files#diff-adb1e6d23c5a4c850b9b760b27a3d557b341c5c438526f607068e4d1f9d00352R190) the `zone`, which is only available after the zone has been matched.

Reply:
I see. Actually, if we store the zone in the status of the GSLB then the finalizer can use the status of the last reconciliation, which is already present on the GSLB resource.